src/main/java/org/commonjava/indy/service/tracking/jaxrs/AdminResource.java

```diff
@@ -169,24 +171,36 @@ public Response initRecord(
+            @Parameter( name = "packageType", in = QUERY, description = "The package type of the repository.",
+                        example = "maven, npm, generic-http", required = true ),
+            @Parameter( name = "type", in = QUERY, description = "The type of the repository.",
+                        example = "group, remote, hosted", required = true ) } )
     @APIResponse( responseCode = "200", description = "Tracking record" )
     @APIResponse( responseCode = "404", description = "No such tracking record" )
     @Path( "/{id}/artifactRecord/{path: (.*)}" )
     @GET
-    public Response recordArtifact(
-            @Parameter( description = "User-assigned tracking session key", in = PATH, required = true )
-            @PathParam( "id" ) final String id, @PathParam( "path" ) String path,
-            @QueryParam( "packageType" ) String packageType, @QueryParam( "type" ) String type,
-            @QueryParam( "name" ) String name, @QueryParam( "originalUrl" ) String originalUrl,
-            @QueryParam( "size" ) long size, @QueryParam( "md5" ) String md5, @QueryParam( "sha1" ) String sha1,
-            @QueryParam( "sha256" ) String sha256, @Context final UriInfo uriInfo )
+    public Response recordArtifact( @Parameter( in = PATH, required = true ) @PathParam( "id" ) final String id,
+                                    @Parameter( in = PATH, required = true ) @PathParam( "path" ) String path,
+                                    @Parameter( in = QUERY, required = true ) @QueryParam( "packageType" )
+                                    String packageType,
+                                    @Parameter( in = QUERY, required = true ) @QueryParam( "type" ) String type,
+                                    @Parameter( in = QUERY, required = true ) @QueryParam( "name" ) String name,
+                                    @QueryParam( "originalUrl" ) String originalUrl, @QueryParam( "size" ) long size,
+                                    @QueryParam( "md5" ) String md5, @QueryParam( "sha1" ) String sha1,
+                                    @QueryParam( "sha256" ) String sha256, @Context final UriInfo uriInfo )
     {
-        TrackedContentEntry contentEntry = new TrackedContentEntry( new TrackingKey( id ), new StoreKey( packageType,
-                                                                                                         StoreType.valueOf(
-                                                                                                                 type ),
-                                                                                                         name ),
-                                                                    AccessChannel.NATIVE, originalUrl, path,
-                                                                    StoreEffect.DOWNLOAD, size, md5, sha1, sha256 );
+        final StoreType st = StoreType.get( type );
+        if ( null == packageType || null == type || null == st )
```

Comment:
@ligangty, just add st check here, for package type, it will check when constructing StoreKey.

Reply:
ok. So LGTM